typing/ctype.ml

```diff
@@ -5697,3 +5697,31 @@ let immediacy env typ =
       else
         Type_immediacy.Always
   | _ -> Type_immediacy.Unknown
+
+(* For use with ocamldebug *)
+type global_state =
+  { current_level : int ref;
+    nongen_level : int ref;
+    global_level : int ref;
+  }
+
+let global_state : global_state =
+  { current_level;
+    nongen_level;
+    global_level;
+  }
+
+let print_global_state fmt { current_level; nongen_level; global_level } =
+  let record_fields =
+    [ "current_level", current_level;
+      "nongen_level", nongen_level;
+      "global_level", global_level;
+    ]
+  in
+  !Oprint.out_value
+    fmt
+    (Oval_record
+      (List.map
+        (fun (printed_name, value) ->
+          Outcometree.Oide_ident { printed_name }, Outcometree.Oval_int !value)
+        record_fields))
```

Comment:
This makes `Ctype` depend on `Oprint` and `Outcometree`, so the build is broken (at least the makefile one - I didn't try dune).

You can fix this by running `make depend` and committing the updated `.depend` file.  But this feels like a somewhat heavy-weight solution.  Perhaps just print directly instead:
```ocaml
  let record_fields =
     ...
  in
  let print_fields fmt fields =
    List.iter (fun (s,r) -> Format.fprintf fmt "%s = %d;@;" s !r) fields
  in
  Format.fprintf fmt "@[<1>{@;%a}@]" print_fields record_fields
```

Reply:
I agree, it seems worse to introduce this dependency than just printing the fields directly (thanks for the suggestion). The output is now:

```
(ocd) p Ctype.global_state
Ctype.global_state: Ctype.global_state =
  { current_level = 2; nongen_level = 2; global_level = 2; }
```